### PR TITLE
Fixed #808

### LINKED ISSUE #808: Global settings changes should only affect new pages or when an existing page is reloaded
### Description: 
Global settings changes should only affect new pages or when an existing page is reloaded. The current behaviour is quite different from desktop. On Desktop this is what happens
1. Visit sites keep default shields settings
2. Block all cookies from global settings
3. Switch to the tab visited in step 1, page doesn't reload but shields shows cookies blocked because global settings is changed, only reload actually causes the global settings to take affect via shields. There is no voluntary reload triggered for any page both active/inactive(background)

But on iOS the current behaviour is

1. Visit sites and keep default shields settings
2. Block all cookies from  settings
3. Active tab reloads without user interaction
4. Disable Block all cookies from settings
5. All open tabs reload to make way for the settings change

**Expected result:**
Behaviour needs to be same on all platforms.

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
All the time

**Brave Version:** 
1.7.3

**Device details:** 
All

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information 
#806

### PR DESCRIPTION
**Pull request by danishjafri88**
## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [ ] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New files have MPL-2.0 license header.

## Reviewer Checklist:

- [ ] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [ ] Issues are assigned to at least one epic.
- [ ] Issues include necessary QA labels:
  - [ ] `QA/(Yes|No)`
  - [ ] `release-notes/(include|exclude)`
  - [ ] `bug` / `enhancement`
- [ ] Necessary security reviews have taken place.
- [ ] Adequate test coverage exists to prevent regressions.
- [ ] Adaquate test plan exists for QA to validate (if applicable)

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -2922,15 +2922,7 @@ extension BrowserViewController: PreferencesObserver {
...
-            if Preferences.Privacy.blockAllCookies.value {
-                tabManager.reset()
-            } else {
-                tabManager.allTabs.forEach {
-                    if let url: URL = $0.webView?.url {
-                        $0.loadRequest(PrivilegedRequest(url: url) as URLRequest)
-                    }
-                }
-            }
+            tabManager.reset()
```
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController.swift:2925]: So the only change here is that we don't reload all tabs on cookies toggle?

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -2922,15 +2922,7 @@ extension BrowserViewController: PreferencesObserver {
              Preferences.Shields.fingerprintingProtection.key:
             tabManager.allTabs.forEach { $0.webView?.reload() }
```
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController.swift:2923]: Shouldn't we stop reloading tabs for other global settings too?

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (danishjafri88) [reply Client/Frontend/Browser/BrowserViewController.swift:2925]: yeah

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (danishjafri88) [reply Client/Frontend/Browser/BrowserViewController.swift:2923]: We should ideally, we could reuse the same reset() for all such cases. Should I do it in this PR itself?

Comment (iccub) [reply Client/Frontend/Browser/BrowserViewController.swift:2923]: No, please add a ticket for it, or maybe there;s something added already, like https://github.com/brave/brave-ios/issues/825

Comment (danishjafri88) [reply Client/Frontend/Browser/BrowserViewController.swift:2923]: great

**Review by iccub — APPROVED** [review]

**Pull request merged**